src/compiler/types.ts

```diff
@@ -5894,7 +5908,13 @@ namespace ts {
 
     export enum ModuleResolutionKind {
         Classic  = 1,
-        NodeJs   = 2
+        NodeJs   = 2,
+        // Starting with node12, node's module resolver has significant departures from tranditional cjs resolution

```

Comment:
Does this comment show up in the quick info of our API? If not, I think we could do with a shorter comment for future maintainers.

Reply:
No, it doesn't show up in our public API - it's more of a rationale comment than anything else.